### PR TITLE
Row similarity

### PR DESCRIPTION
**Pull request by nrposner**

### CODE REVIEW

File: utils/linkage.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import textdistance as td
-import usaddress
+
+# import usaddress
```
Comment (averyschoen) [inline utils/linkage.py]: this needs to be uncommented presumably for the other functions

File: utils/linkage.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
 """
 Module for performing record linkage on state campaign finance dataset
 """
+import numpy as np
```
Comment (averyschoen) [inline utils/linkage.py:8]: put these with the other import statements

File: utils/linkage.py [hunk 3]
```diff
@@ -44,46 +47,100 @@ def calculate_string_similarity(string1: str, string2: str) -> float:
     return float(td.jaro_winkler(string1.lower()[::-1], string2.lower()[::-1]))
 
 
-def get_street_from_address_line_1(address_line_1: str) -> str:
-    """Given an address line 1, return the street name
+def calculate_row_similarity(
```
Comment (averyschoen) [inline utils/linkage.py:99]: we will discuss this in our meeting tomorrow morning

**Review by averyschoen — CHANGES_REQUESTED** [review]

File: utils/linkage.py (hunk 1 above)
Comment (nrposner) [reply utils/linkage.py]: Uncommented and put the address function back in to prevent linter errors.

File: utils/linkage.py [hunk 4]
```diff
@@ -44,6 +46,105 @@ def calculate_string_similarity(string1: str, string2: str) -> float:
...
+    The length of the weights vector must be the same as
+    the number of selected columns.
+
+    This version is slow and not optimized, and will be
+    revised in order to make it more efficient. It
+    exists as to provide basic functionality. Once we have
+    the comparison function locked in, using .apply will
+    likely be easier and more efficient.
+
+    >>> d = {
```
Comment (averyschoen) [inline utils/linkage.py]: lets shift to testing this function per slack convo: https://uchicagodsi.slack.com/archives/C06ADK4FXNX/p1707149767586029

**Review by averyschoen — CHANGES_REQUESTED** [review]

**Review by averyschoen — APPROVED** [review]

**Pull request merged**